### DISCUSSION TITLE
Lesson 7 - Storage: command "cast storage 0x... 2" returns 0x...

### DISCUSSION THREAD

**Discussion by JacobHomanics** [Cyfrin/foundry-full-course-cu#423] [Q&A]
Hey guys, so I'm following along and going through Lesson 7 and am on the Storage section.

Pat goes through a list of commands in different terminals:
Terminal 1 -> `anvil`
Terminal 2 -> `forge script script/DeployFundMe.s.sol --rpc-url http://127.0.0.1:8545 --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 --broadcast`
Terminal 3 -> `cast storage $AddressPulledFromTerminal2 2`

and he gets back a random hex value, meanwhile I get back a hex value that is all zeroes. I am just trying to understand why that is happening. I have also tried `cast storage $AddressPulledFromTerminal2 0`, `cast storage $AddressPulledFromTerminal2 1`, `cast storage $AddressPulledFromTerminal2 2`, `cast storage $AddressPulledFromTerminal2 3`. They all return a hex value that is all zeroes. 

Here is what my FundMe.sol looks like:
```
contract FundMe {
    using PriceConverter for uint256;

    mapping(address => uint256) private s_addressToAmountFunded;
    address[] private s_funders;

    address private immutable i_owner;
    uint256 public constant MINIMUM_USD = 5e18;

    AggregatorV3Interface private s_priceFeed;
```
Here is what my DeployFundMe.s.sol looks like:
```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Script} from "forge-std/Script.sol";
import {FundMe} from "../src/FundMe.sol";
import {HelperConfig} from "./HelperConfig.s.sol";

contract DeployFundMe is Script {
    function run() public returns (FundMe) {
        HelperConfig config = new HelperConfig();
        address priceFeedAddr = config.activeNetworkConfig();

        vm.startBroadcast();
        FundMe fundMe = new FundMe(priceFeedAddr);
        vm.stopBroadcast();
        return fundMe;
    }
}
```

Here is what the command `forge inspect FundMe storagelayout` shows for my storage array:
```
"storage": [
    {
      "astId": 43139,
      "contract": "src/FundMe.sol:FundMe",
      "label": "s_addressToAmountFunded",
      "offset": 0,
      "slot": "0",
      "type": "t_mapping(t_address,t_uint256)"
    },
    {
      "astId": 43142,
      "contract": "src/FundMe.sol:FundMe",
      "label": "s_funders",
      "offset": 0,
      "slot": "1",
      "type": "t_array(t_address)dyn_storage"
    },
    {
      "astId": 43150,
      "contract": "src/FundMe.sol:FundMe",
      "label": "s_priceFeed",
      "offset": 0,
      "slot": "2",
      "type": "t_contract(AggregatorV3Interface)45"
    }
  ],
```

Let me know if more information is required to figure out whats going on! Thank you.

EDIT: I went ahead and wrote a test for my HelperConfig to make sure the priceFeed address wasn't being set to ZeroAddress, and can confirm priceFeed is set to a unique address
<img width="1395" alt="Screenshot 2023-07-23 at 8 52 56 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/32080359/f563736c-c39e-4b4f-96bc-7d3f0c344104">

**Comment by john-lawniczak** [comment]
I don't have the answer for you, but.. already writing tests! 

PretttTTTTTTy good.

**Comment by ghost** [comment]
I think the command only works on contracts that are verified on Etherscan.

**Comment by gilish-tech** [comment]
The problem is you did not pass a private-key
```solidity
vm.startBroadcast(uint256(0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80));
```

**Reply by EngrPips** [reply]
Interesting
